smoketests/tests/csharp_module.py

```diff
@@ -0,0 +1,50 @@
+from .. import run_cmd, STDB_DIR, requires_dotnet, spacetime
+import unittest
+import tempfile
+from pathlib import Path
+import shutil
+import subprocess
+
+
+@requires_dotnet
+class CreateProject(unittest.TestCase):
+    def test_build_csharp_module(self):
+        """
+        Ensure that the CLI is able to create and compile a csharp project. This test does not depend on a running spacetimedb instance. Skips if dotnet 8.0 is not available
+        """
+
+        bindings = Path(STDB_DIR) / "crates" / "bindings-csharp"
+        codegen = bindings / "Codegen"
+        runtime = bindings / "Runtime"
+
+        try:
+
+            run_cmd("dotnet", "workload", "install", "wasi-experimental")
+            run_cmd("dotnet", "pack", cwd=codegen, capture_stderr=True)
+            run_cmd("dotnet", "pack", cwd=runtime, capture_stderr=True)
+
+            with tempfile.TemporaryDirectory() as tmpdir:
+                spacetime("init", "--lang=csharp", tmpdir)
+
+                codegen_bin = codegen / "bin" / "Release"
+                runtime_bin = runtime / "bin" / "Release"
+
+                csproj = Path(tmpdir) / "StdbModule.csproj"
+                with open(csproj, "r") as f:
+                    contents = f.read()
+
+                contents = contents.replace(
+                    "</PropertyGroup>",
+                    f"""<RestoreAdditionalProjectSources>{codegen_bin.absolute()};{runtime_bin.absolute()}</RestoreAdditionalProjectSources>
```

Comment:
As mentioned in comment, I'd opt for `RestoreSources` rather than `RestoreAdditionalProjectSources` here to ensure that we're getting and testing the local package instead of silently getting one from nuget.org.

Reply:
Unfortunately we have additional dependencies that do not resolve when we do that.

Reply:
Oh, I know, I'll patch the version!

Reply:
You should be able to add nuget URL explicitly as the last element.

Reply:
Wait, that is a far better solution than what I just did. I will do that instead.